Updated Java getting started steps with sample app (#5529)
influxdata/docs-v2: content/influxdb/cloud-serverless/reference/client-libraries/v3/java.md

The documentation for this client library is available on GitHub.

<a href="https://github.com/InfluxCommunity/influxdb3-java" target="_blank" class="btn github">InfluxDB v3 Java client library</a>

## Installation

### Maven

> ```xml

Reviewer: ~~~suggestion
### Maven

```xml
~~~

<dependency>
  <groupId>com.influxdb</groupId>
  <artifactId>influxdb3-java</artifactId>
  <version>RELEASE</version>
</dependency>
```

### Gradle

> ```groovy

Reviewer: ~~~suggestion
### Gradle

```groovy
~~~

implementation group: 'com.influxdb', name: 'influxdb3-java', version: 'latest.release'
```

> ## Sample Code

Reviewer 2: ```suggestion
## Example: write and query data

The following example shows how to use `influxdb3-java` to write and query data stored in {{% product-name %}}:
```

```java
public class HelloInfluxDB {

>   private static final String HOST_URL = "https://eu-central-1-1.aws.cloud2.influxdata.com"; // URL information is present in the cloud portal

Reviewer: The docs do one of the following to automatically populate cloud URLs:

- If the user is referred to the docs from their InfluxDB Cloud dashboard, we detect the referrer and automatically set their URL.
- The user can manually select their region and we'll automatically populate the URL.

```suggestion
  private static final String HOST_URL = "https://{{< influxdb/host >}}";
```

  private static final String DATABASE = "java"; // Database name is the bucket name present in the cloud portal
  private static final char[] API_TOKEN = "API_TOKEN".toCharArray(); // Avoid hard-coding API_TOKEN in production. It is present in the cloud portal.

>   // Authenticate, Write and Query data from the serverless InfluxDB

Reviewer 2: ```suggestion
  // Create a client instance, and then write and query data in InfluxDB Cloud Serverless.
```

  public static void main(String[] args) {
    try (InfluxDBClient client = InfluxDBClient.getInstance(HOST_URL, API_TOKEN, DATABASE)) {
      writeData(client);
      queryData(client);
    }
    catch (Exception e) {
      System.err.println("An error occurred while connecting with the serverless InfluxDB!");
      e.printStackTrace();
    }
  }

>   // Write sample measurement using Point class API

Reviewer 2: ```suggestion
  // Use the Point class to construct time series data.
  // Call client.writePoint to write the point as line protocol to your bucket.
```

                       .setTag("location", "London")
                       .setField("value", 30.01)
                       .setTimestamp(Instant.now().minusSeconds(10));
    try {
      client.writePoint(point);
      System.out.println("Data written to the bucket.");
    }
    catch (Exception e) {
      System.err.println("Failed to write data to the bucket.");
      e.printStackTrace();
    }
  }

  // Query the latest 10 measurements using SQL
  private static void queryData(InfluxDBClient client) {
    System.out.printf("--------------------------------------------------------%n");
    System.out.printf("| %-8s | %-8s | %-30s |%n", "location", "value", "time");
    System.out.printf("--------------------------------------------------------%n");

    String sql = "select time,location,value from temperature order by time desc limit 10";
    try (Stream<Object[]> stream = client.query(sql)) {
      stream.forEach(row -> System.out.printf("| %-8s | %-8s | %-30s |%n", row[1], row[2], row[0]));
    }
    catch (Exception e) {
      System.err.println("Failed to query data from the bucket.");
      e.printStackTrace();
    }
  }
}

> ```

Reviewer: Comment on this code example as a whole: I think it might be better to split this up and explain each interaction with the client:

- Set up authentication
- Write data to InfluxDB
- Query data from InfluxDB

Then show the full `.jar` file after everything has been explained.

> ## Running the program

Reviewer 2: ```suggestion
## Run the program to write and query data
```

> Build the project and then run the executable .jar file with this [JVM Flag](https://arrow.apache.org/docs/java/install.html). 

Reviewer 2: ```suggestion
1. Build the project using Maven or Gradle to create an executable JAR file.
2. Run the following command to execute the JAR:
```

> <br/><br/> 

Reviewer 2: ```suggestion

```

> **Example:**

Reviewer 2: ```suggestion
```

```sh

>   java --add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED -jar target/{your-jar-file}.jar

Reviewer 2: ```suggestion
  java --add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED -jar target/PROJECT_NAME.jar
```

> ```

Reviewer: ~~~suggestion
```

## Run your program

Build the project and then run the executable `.jar` file with the [JVM `--add-opens` flag](https://arrow.apache.org/docs/java/install.html). For example:

{{% code-placeholders "YOUR_JAR_FILE" %}}

```sh
java --add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED -jar target/YOUR_JAR_FILE.jar
```

{{% /code-placeholders %}}

Replace {{% code-placeholder-key %}}`YOUR_JAR_FILE`{{% /code-placeholder-key %}} with the name of your `.jar` file.
~~~
